src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -408,14 +467,44 @@ private void CreateNewIndex(Directory dir)
                     //unlock it!
                     IndexWriter.Unlock(dir);
                 }
+
                 //create the writer (this will overwrite old index files)
                 writer = new IndexWriter(dir, FieldAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED);
+                if (_directory is ExamineDirectory examineDirectory)
+                {
+                 
+                    if (examineDirectory.GetDeletionPolicy() != null)
+                    {
+                        writer = new IndexWriter(dir, FieldAnalyzer, true, examineDirectory.GetDeletionPolicy(),
```

Comment:
The original writer is not being disposed of when it's replaced.

Reply:
@nzdev true I forgot to dispose that, fixing :)